server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -958,23 +956,47 @@ private void checkForHeldServer(SortedMap<TServerInstance,TabletServerStatus> ts
       }
     }
 
-    private long balanceTablets() {
-      BalanceParamsImpl params = BalanceParamsImpl.fromThrift(tserverStatusForBalancer,
-          tserverStatus, migrationsSnapshot());
-      long wait = tabletBalancer.balance(params);
+    private long balanceTablets(boolean balanceUserTables) {
 
-      for (TabletMigration m : checkMigrationSanity(tserverStatusForBalancer.keySet(),
-          params.migrationsOut())) {
-        KeyExtent ke = KeyExtent.fromTabletId(m.getTablet());
-        if (migrations.containsKey(ke)) {
-          log.warn("balancer requested migration more than once, skipping {}", m);
+      Map<DataLevel,Set<KeyExtent>> partitionedMigrations =
+          new HashMap<>(DataLevel.values().length);
+      migrationsSnapshot().forEach(ke -> {
+        partitionedMigrations.computeIfAbsent(DataLevel.of(ke.tableId()), f -> new HashSet<>())
+            .add(ke);
+      });
+
+      BalanceParamsImpl params = null;
+      long wait = 0;
+      for (DataLevel dl : new DataLevel[] {DataLevel.ROOT, DataLevel.METADATA, DataLevel.USER}) {
+        Set<KeyExtent> migrationsForLevel = partitionedMigrations.get(dl);
+        if (migrationsForLevel == null) {
+          continue;
+        }
+        if (dl == DataLevel.USER && !balanceUserTables) {
+          log.debug("not balancing user tablets because there are {} unhosted tablets",
+              notHosted());
```

Comment:
Could drop the boolean passed to the method and call notHosted here.  Then the log stmt can print the value that was used to make the decision.

```suggestion
        int  notHostedCount = notHosted();
        if (dl == DataLevel.USER && notHostedCount > 0) {
          log.debug("not balancing user tablets because there are {} unhosted tablets",
              notHostedCount);
```

Reply:
Implemented in 26a4939